### PR TITLE
AP_HAL_LINUX: Add support for Raspberry Pi 4

### PR DESCRIPTION
**Pull request by patrickelectric**
The Raspberry Pi 4 uses a new BCM CPU, the BCM2711 with
a peripheral base address of 0xFE000000

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: libraries/AP_HAL_Linux/Util_RPI.cpp [hunk 1]
```diff
@@ -57,7 +58,10 @@ int UtilRPI::_check_rpi_version()
     // Attempting old method if the version couldn't be read with the new one.
     hw = Util::from(hal.util)->get_hw_arm32();
 
-    if (hw == UTIL_HARDWARE_RPI2) {
+    if (hw == UTIL_HARDWARE_RPI4) {
+        printf("Raspberry Pi 24 with BCM2711!\n");
```
Comment (amilcarlucas) [inline libraries/AP_HAL_Linux/Util_RPI.cpp]: Raspberry Pi 4 with BCM2711!\n

Comment (patrickelectric) [reply libraries/AP_HAL_Linux/Util_RPI.cpp]: Ops.

**Comment by lucasdemarchi** [comment]
Could you probe the spi bus clk and check it is constant?  I've heard there's a bug in which the spi clk may drop when the cpu freq drops.

**Comment by jaxxzer** [comment]
> Could you probe the spi bus clk and check it is constant? I've heard there's a bug in which the spi clk may drop when the cpu freq drops.

@lucasdemarchi We are testing with some spi sensors, and they are working. We will check this for you in our testing, but I think it is not something that needs to be addressed before this is merged.

**Pull request merged**